spec/manageiq/messaging/client_spec.rb

```diff
@@ -0,0 +1,116 @@
+require "spec_helper"
+
+describe ManageIQ::Messaging::Client do
+  before do
+    module ManageIQ::Messaging::Test
+      class Client < ManageIQ::Messaging::Client
+        def initialize(options); end
+        def close; end
+        def publish_message_impl(args); end
+        def publish_topic_impl(args); end
+        def publish_messages_impl(args); end
+        def subscribe_messages_impl(args); end
+        def subscribe_topic_impl(args); end
+        def subscribe_background_job_impl(args); end
+      end
+    end
```

Comment:
Since this class does not collide with anything in the gem, and since there doesn't seem to be any state preserved in it between tests, you can probably just make this a top-level test class in the spec_helper.  Then you don't need to destroy it in the after either.